### ISSUE TITLE
Duplicate Songs in library and playlist while files are normal

### ISSUE THREAD

**Issue by ReIaxo** [strawberrymusicplayer/strawberry#1311]
- [x] I have checked the [FAQ](https://wiki.strawberrymusicplayer.org/wiki/FAQ) for answers.
- [x] I have checked the [Changelog](https://github.com/strawberrymusicplayer/strawberry/blob/master/Changelog) that the issue is not already fixed. 
- [x] I believe this issue is a bug, and not a general technical issue, question or feature requests that can be discussed on the [forum](https://forum.strawberrymusicplayer.org/). 

**Describe the bug**
There are duplicate entries of songs in the playlist when I double click an artist in library. There are also duplicates shown in the library tree.
One is shown with file type MP3 and one with Ogg Vorbis.
I am not really sure if this is an issue with the file or with Strawberry.
It does not happen for all MP3 files.

**To Reproduce**
Add some of this special files to he library and add all songs of the artist to a playlist.

**Expected behavior**
Only one occurence of the song shown on library tree and playlist. I think the MP3 one.

**Screenshots:**
Songs from Album Stone Sour are duplicated. Even if they are only once in the file system.
![Bildschirmfoto vom 2023-11-08 09-02-34](https://github.com/strawberrymusicplayer/strawberry/assets/44195536/76217bce-04d8-4636-b87d-9551386bf0ff)
![Bildschirmfoto vom 2023-11-08 09-05-24](https://github.com/strawberrymusicplayer/strawberry/assets/44195536/df5d6428-3de3-4135-adfa-501b6d271e4b)

**System Information:**
 - Operating system: Ubuntu 23.04
 - Strawberry Version: 1.0.17

**Additional context**
I think i am not allowed to share the file here.

**Comment by jonaski** [comment]
You probably have a copy of the same songs in Ogg Vorbis.
Right click on the playlist header and select "File Name" to see the file paths for the songs.
Right click on the Ogg Vorbis songs and select "Show in file browser".

**Comment by ReIaxo** [comment]
I think it is the exact same file.
![Bildschirmfoto vom 2023-11-08 22-14-45](https://github.com/strawberrymusicplayer/strawberry/assets/44195536/580ede36-be6a-4f6c-aeeb-bda1113d220e)

And sorry for messing up the versions. I mean 1.0.17. And updated to 1.0.21 via apt, same behavior.

**Comment by jonaski** [comment]
I find it very unlikely that a song is added twice to the collection from the same file as two different audio formats, so I'm curios to how this has happened. Was it like this after a new collection scan? do you got multiple collection directories?
If you want, you can send me the database and I can have a look at it.
Otherwise, if you don't care about losing playlists or rating/playcounts, suggest to remove the collection directory and re-add it and see if the problem disappears.
Otherwise manually fixing the duplicates in the database is a possibility: https://wiki.strawberrymusicplayer.org/wiki/Find_and_delete_duplicates_from_the_collection_database

**Comment by ReIaxo** [comment]
Okay, checked it with the same files on another machine (Win10) and after removing the strawberry.db. Both works and the songs only show up once. I think things were messed up by migrating from clementine to strawberry a year ago. I will live with loosing my listens count and stuff. The worst thing about it is, that I have to add all my 100+ playlists one by one again. Is there maybe a feature in sight to add multiple at once? Or a workaround script with manual sqlite calls?

**Comment by jonaski** [comment]
If it's only a few albums that's duplicated, then you should be able to simply manually deleting those duplicates or just delete the albums from the database and do a full rescan.
Something like:
`delete from songs where artist = 'artist' and album ='album';`

**Comment by Ji59** [comment]
I also encountered this problem in Strawberry `1.2.18` on my Fedora 43 system.
I noticed this issue after adding new albums from Bandcamp with FLAC files, and then deleting existing songs in different albums.
I see duplicates of the same song even with the same file type, as could be seen in image below.
Full collection rescan didn't help.

<img width="2560" height="1440" alt="Image" src="https://github.com/user-attachments/assets/969a95c8-f697-4085-b997-f0741e2f06d4" />.

After running `find . -name '*.wav' -or -name '*.flac' -or -name '*.ogg' -or -name '*.m4a' -or -name '*.mp3' -or -name '*.opus' | wc -l`, I get `4722`.
So there should be `6511 - 4722 = 1789` duplicates.

I use Syncthing to synchronize the music files with my other PC (with Strawberry) and Android phone, so it could be causing some problems.

I tried to follow the steps listed in [Find and delete duplicates from the collection database](https://wiki.strawberrymusicplayer.org/wiki/Find_and_delete_duplicates_from_the_collection_database) wiki page.
I wasn't able to run `DELETE FROM songs_fts;` with error: `Parse error: no such table: songs_fts` and indeed, there is no such table in my database.
But after reopening Strawberry and doing full collection rescan to be sure, everything works.
